File: groupAnagram/ga.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        dic = {}
        i = 0
        for s in strs:
            l=list(s)
            l.sort()
            news="".join(l)
            if news in dic.keys():
                dic[news].append(s)
            else:
                dic[news] = [s]
        return list(dic.values())

File: groupAnagram/ga.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String, List<String>> res = new HashMap<String, List<String>>();
        for(String s: strs){
            char[] l = s.toCharArray();
            Arrays.sort(l);
            String news = String.valueOf(l);
            if(!res.containsKey(news)){
                res.put(news, new ArrayList<String>());
            }
            res.get(news).add(s);
        }
        return new ArrayList<List<String>>(res.values());
    }
}